Wizardry/Core/SkillSys/MiscSkillEffects/PostActionSkills/SoulSap.c

```diff
@@ -0,0 +1,85 @@
+#include "common-chax.h"
+#include "skill-system.h"
+#include "map-anims.h"
+#include "debuff.h"
+#include "constants/skills.h"
+
+STATIC_DECLAR void ExecSkillSoulSapEffectAnim(ProcPtr proc)
+{
+    struct Unit * unit = gActiveUnit;
+
+    BmMapFill(gBmMapMovement, -1);
+    BmMapFill(gBmMapRange, 0);
+    GenerateUnitStandingReachRange(unit, 0b110);
+
+    InitTargets(unit->xPos, unit->yPos);
+    ForEachUnitInRange(AddUnitToTargetListIfNotAllied);
+
+#if 0
+#endif
+}
+
+STATIC_DECLAR void SkillSoulSapPostAnimEffect(ProcPtr proc)
+{
+    int i;
+    int heal_amt = 0;
+    struct Unit * unit = gActiveUnit;
+
+    for (i = 0; i < GetSelectTargetCount(); i++)
+    {
+        struct SelectTarget * starget = GetTarget(i);
+        struct Unit * tunit = GetUnit(starget->uid);
+
+        int max_hp = GetUnitMaxHp(tunit);
+#if defined(SID_SoulSap) && (COMMON_SKILL_VALID(SID_SoulSap))
+        int perc = SKILL_EFF0(SID_SoulSap);
+#else
+        int perc = 10;
+#endif
+        int damage_amt = Div(max_hp * perc, 100);
+
+        // Make sure enemy units don't die from this skill
+        if (tunit->curHP <= damage_amt)
+        {
+            tunit->curHP = 1;
+            heal_amt += tunit->curHP - 1;
```

Comment:
bug

Reply:
Took me a while to see it because you don't elaborate, but yeah will fix